mm/ktsan/sync_atomic.c

```diff
@@ -68,6 +71,37 @@ do {									\
 	kt_spin_unlock(&sync->tab.lock);					\
 } while (0)
 
+#else /* KT_DEBUG */
+
+#define KT_ATOMIC_OP(op, ad, mo, read, write)				\
+do {									\
+	kt_tab_sync_t *sync;						\
+									\
+	sync = kt_sync_ensure_created(thr, pc, (ad));			\
+									\
+	if ((mo) == ktsan_memory_order_acquire ||			\
```

Comment:
I think it's better to move this part:

+	if ((mo) == ktsan_memory_order_acquire ||			\
+	    (mo) == ktsan_memory_order_acq_rel) {			\
+		kt_clk_acquire(&thr->clk, &sync->clk);			\
+		kt_thread_fence_no_ktsan(ktsan_memory_order_acquire);	\
+	} else if (read) {						\
+		kt_clk_acquire(&thr->acquire_clk, &sync->clk);		\
+	}	

into a separate function. And the same for release part.

Then you can have a single macro for KT_DEBUG=0/1.